crates/turbopack-ecmascript/src/transform/mod.rs

```diff
@@ -215,9 +222,15 @@ impl EcmascriptInputTransform {
                     ..Default::default()
                 };
                 let p = std::mem::replace(program, Program::Module(Module::dummy()));
+                let hash = {
+                    let mut hasher = DefaultHasher::new();
+                    p.hash(&mut hasher);
```

Comment:
That sounds pretty expensive. Do we need to hash the whole AST?

Reply:
What I actually need is the hash of the source code. Would it be better?

Reply:
We can pass that via TransformContext

Reply:
But how is the transform using that? Would a comment change have effects on the transform?

Reply:
It's used to generate a hash for the file.

Related: https://github.com/swc-project/plugins/pull/163